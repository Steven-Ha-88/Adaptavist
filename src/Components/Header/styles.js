import styled, { createGlobalStyle } from "styled-components";

export const Nav = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: 80px;
  background-color: rgb(250, 105, 77);
`;

export const Brand = styled.h1`
  font-family: "Roboto", sans-serif;
  font-weight: 500;
  color: white;
  margin-left: 10px;
  font-size: 30px;
  @media (max-width: 500px) {
    font-size: 20px;
  }
`;

export const Globalstyle = createGlobalStyle`
  body {
   margin: 0; 
  }
`;
