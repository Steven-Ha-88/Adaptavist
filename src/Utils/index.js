import {
  SVG1d,
  SVG1n,
  SVG2d,
  SVG2n,
  SVG3d,
  SVG3n,
  SVG4d,
  SVG4n,
  SVG9d,
  SVG9n,
  SVG10d,
  SVG10n,
  SVG11d,
  SVG11n,
  SVG13d,
  SVG13n,
  SVG50d,
  SVG50n
} from "../images/index";

export const findTestAttr = (component, attribute) => {
  const wrapper = component.find(`[data-test="${attribute}"]`);
  return wrapper;
};

export const toCelcius = kel => {
  const temp = Math.round(kel - 273.15);
  return temp;
};

export const weatherImg = term => {
  switch (term) {
    case "01d":
      return SVG1d;
    case "01n":
      return SVG1n;
    case "02d":
      return SVG2d;
    case "02n":
      return SVG2n;
    case "03d":
      return SVG3d;
    case "03n":
      return SVG3n;
    case "04d":
      return SVG4d;
    case "04n":
      return SVG4n;
    case "09d":
      return SVG9d;
    case "09n":
      return SVG9n;
    case "10d":
      return SVG10d;
    case "10n":
      return SVG10n;
    case "11d":
      return SVG11d;
    case "11n":
      return SVG11n;
    case "13d":
      return SVG13d;
    case "13n":
      return SVG13n;
    case "50d":
      return SVG50d;
    case "50n":
      return SVG50n;
    default:
      return null;
  }
};
