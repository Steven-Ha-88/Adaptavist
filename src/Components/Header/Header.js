import React from "react";
import { Nav, Brand, Globalstyle } from "./styles";
import SearchBar from "../SearchBar/SearchBar";

const Header = () => (
  <Nav data-test="header">
    <Globalstyle />
    <Brand data-test="logo"> My Weather App </Brand>
    <SearchBar row />
  </Nav>
);

export default Header;
