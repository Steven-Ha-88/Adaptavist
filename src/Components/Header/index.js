import React from "react";
import { Link } from "react-router-dom";
import { Nav, Brand, Globalstyle } from "./styles";
import SearchBar from "../SearchBar";

const Header = () => {
  return (
    <Nav data-test="header">
      <Globalstyle />
      <Link to="/" style={{ textDecoration: "none" }}>
        <Brand data-test="logo"> My Weather App </Brand>
      </Link>
      <SearchBar data-test="search" row />
    </Nav>
  );
};

export default Header;
