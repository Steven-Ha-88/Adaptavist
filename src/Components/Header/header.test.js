import React from "react";
import { shallow } from "enzyme";
import Header from "./index";
import { findTestAttr } from "./../../Utils";
import toJson from "enzyme-to-json";

const setUp = (props = {}) => {
  const component = shallow(<Header {...props} />);
  return component;
};

describe("Header Component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("mounts", () => {
    expect(component.exists()).toEqual(true);
  });

  it("Should render without errors", () => {
    const wrapper = findTestAttr(component, "header");
    expect(wrapper.length).toBe(1);
  });

  it("Should render a logo", () => {
    const wrapper = findTestAttr(component, "logo");
    expect(wrapper.length).toBe(1);
  });

  it("Should render search bar component", () => {
    const wrapper = findTestAttr(component, "search");
    expect(wrapper.length).toBe(1);
  });

  it("Should match snapshot", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
