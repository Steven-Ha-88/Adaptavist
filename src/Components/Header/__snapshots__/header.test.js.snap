// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Should match snapshot 1`] = `
<styled.div
  data-test="header"
>
  <GlobalStyleComponent />
  <Link
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/"
  >
    <styled.h1
      data-test="logo"
    >
       My Weather App 
    </styled.h1>
  </Link>
  <SearchBar
    data-test="search"
    row={true}
  />
</styled.div>
`;
